lib/OpenQA/WebSockets/Controller/Worker.pm

```diff
@@ -158,76 +158,63 @@ sub _message ($self, $json) {
-
-            my $registered_job_token;
-            my $current_job_state;
-            my @unfinished_jobs = $worker->unfinished_jobs;
-            my $current_job = $worker->job // $unfinished_jobs[0];
-            if ($current_job) {
-                $current_job_id = $current_job->id;
-                $current_job_state = $current_job->state;
-            }
-
-            # log debugging info
-            log_trace("Found job $current_job_id in DB from worker_status update sent by worker $worker_id")
-              if defined $current_job_id;
-            log_trace("Received request has job id: $job_id")
-              if defined $job_id;
-            $registered_job_token = $worker->get_property('JOBTOKEN');
-            log_trace("Worker $worker_id for job $current_job_id has token $registered_job_token")
-              if defined $current_job_id && defined $registered_job_token;
-            log_trace("Received request has token: $job_token")
-              if defined $job_token;
-
-            # skip any further actions if worker just does the one job we expected it to do
-            return undef
-              if ( defined $job_id
-                && defined $current_job_id
-                && defined $job_token
-                && defined $registered_job_token
-                && $job_id eq $current_job_id
-                && (my $job_token_correct = $job_token eq $registered_job_token)
-                && OpenQA::Jobs::Constants::meta_state($current_job_state) eq OpenQA::Jobs::Constants::EXECUTION)
-              && (scalar @unfinished_jobs <= 1);
-
             # give worker a second chance to process the job assignment
             # possible situation on the worker: The worker might be sending a status update claiming it is
             # idle (or has doing that task piled up on the event loop). At the same time a job arrives. The
             # message regarding that job will be processed after sending the idle status. So let's give the
-            # worker another change to process the message about its assigned job.
-            return undef unless $worker_previously_idle;
-
-            log_debug("Rescheduling jobs assigned to worker $worker_id");
-            $worker->reschedule_assigned_jobs([$current_job, @unfinished_jobs]);
+            # worker another chance to process the message about its assigned job.
+            if ($worker_previously_idle) {
+                log_debug("Rescheduling jobs assigned to worker $worker_id");
+                $worker->reschedule_assigned_jobs([$current_job, @unfinished_jobs]);
+            }
         }
         catch {
             # uncoverable statement
             log_warning("Unable to verify whether worker $worker_id runs its job(s) as expected: $_");
         };
 
         # consider the worker idle unless it claims to be broken or work on a job
-        $worker_status->{idle_despite_job_assignment}
-          = !$worker_is_broken && !defined $job_id && defined $current_job_id;
```

Comment:
This variable is no longer set to a false value in some cases. This leaves it completely missing in some cases considering we initially delete it (via `delete $worker_status->{idle_despite_job_assignment}`). However, it shouldn't make a difference because we never distinguish between the absence of this and a false value.